azavea/raster-foundry
# Add 'ingestLocation' field with filtering to scenes

## Overview

This PR adds a new field to the scenes table: `ingestLocation`. It is a nullable field. It's intended use is to represent the location on S3 where the scene is located.

The scene field `status` was removed from the scenes table.

Also included is the ability to filter based on this field. The query parameter, `ingested` is a boolean that will filter for scenes that have been ingested (or not) based on the parameter. If the parameter is not present, scene ingest status is ignored.

This PR also adds filtering for ingest status on the front-end.

## Checklist


## Demo

<img width="528" alt="screen shot 2017-01-20 at 3 47 50 pm" src="https://cloud.githubusercontent.com/assets/2442245/22165253/e661c0a2-df28-11e6-9945-efe5544474f4.png">

## Notes

We don't have data for this field, so testing is somewhat limited.

The wording of the filter may need to be changed, I had trouble thinking of a simple way to phrase the filter sates.

## Testing Instructions

* `vagrant ssh` into the VM
* Start an sbt shell: `./scripts/console app-server ./sbt`
* `mg update` and `mg apply` to drop `status` and add `ingest_location` to the DB
* Exit the sbt shell and start the API
* Test the API via curl:

Should have all scenes:
```
curl -H "Content-Type: application/json" \
     -H "Authorization: Bearer <TOKEN>" \
      http://localhost:9000/api/scenes?ingested=false
```

Should have no scenes:
```
curl -H "Content-Type: application/json" \
     -H "Authorization: Bearer <TOKEN>" \
      http://localhost:9000/api/scenes?ingested=true
```

* Open front-end in a browser
* Ensure that the 'Ingest Status' filter is available within the browse filter pane
* Check that 'Show All' is active
* Change the filter and ensure that url is changed and map (with proper grid cells) reloads to reflect the change

Connects #940
